router/publishfilter.go

```diff
@@ -79,17 +85,17 @@ func newPublishFilter(msg *wamp.Publish) *publishFilter {
 	if blIDs == nil && wlIDs == nil && blMap == nil && wlMap == nil {
 		return nil
 	}
-	return &publishFilter{blIDs, wlIDs, blMap, wlMap}
+	return &simplePublishFilter{blIDs, wlIDs, blMap, wlMap}
 }
 
-// publishAllowed determines if a message is allowed to be published to a
+// PublishAllowed determines if a message is allowed to be published to a
 // subscriber, by looking at any blacklists and whitelists provided with the
 // publish message.
 //
 // To receive a published event, the subscriber session must not have any
 // values that appear in a blacklist, and must have a value from each
 // whitelist.
-func (f *publishFilter) publishAllowed(sub *session) bool {
+func (f *simplePublishFilter) PublishAllowed(sub *Session) bool {
```

Comment:
I do not want to expose the `router.session`.  This is a wrapper around `wamp.Session` and contains items that should remain private to the router.  If you have `PublishAllow()` take a `*wamp.Session` and move the locking outside of the call to `PublishAllowed()`, I think that will accomplish.

I really want to make every effort to not force the user to decide when to read-lock or write-lock a session.  If we make the user do that, then we need to reexamine the design. 

Again, consider implementing client-side filtering.

Reply:
That's right, didn't consider to move the locking out of `PublishAllowed`, will change this one.